src/easylink/pipeline_schema.py

```diff
@@ -37,10 +37,10 @@ def _get_schemas(cls) -> List["PipelineSchema"]:
             PipelineSchema._generate_schema(
                 "development",
                 validate_dummy_input,
-                Step("step_1"),
-                Step("step_2"),
-                Step("step_3"),
-                Step("step_4"),
+                Step("step_1", prev_input=False, input_files=True),
+                Step("step_2", prev_input=True, input_files=False),
+                Step("step_3", prev_input=True, input_files=False),
+                Step("step_4", prev_input=True, input_files=False),
```

Comment:
Just to track my thinking: I expect these flags will go away and be replaced with a graph data structure in the pipeline schema.